File: leetcode/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int i = 0;
        int j = height.size() - 1;
        while(i < j){
            res = max(res, min(height[i], height[j]) * (j - i));
            if(height[j] > height[i]){
                ++i;
            } else {
                --j;
            }
        }
        return res;
    }
};

File: leetcode/11. Container With Most Water.rs
use std::cmp;

impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let mut i = 0;
        let mut j = height.len() - 1;
        let mut res = 0;
        
        while i < j {
            res = cmp::max(res, cmp::min(height[i], height[j]) * (j - i) as i32);
            if height[i] <= height[j] {
                i += 1;
            } else if height[i] > height[j] {
                j -= 1;
            }
        }
        
        return res
    }
}
